tools/OMIcheck/OMIcheck.ps1

```diff
@@ -63,15 +72,31 @@ foreach ($sub in $subs)
                     Write-Host -ForegroundColor Red `t`t $VM.Name ": VM has vulnerable OMI version " $pkgVer
                     if ($upgradeOMI)
                     {
-                        $upgrade = Invoke-AzVMRunCommand -ResourceGroupName $VM.ResourceGroupName -Name $VM.Name -CommandId 'RunShellScript' -ScriptPath $upgradeScriptPath
-                        Write-Host -ForegroundColor Red `t`t $VM.Name ": Result of OMI package upgrade attempt: " $upgrade.Value.Message
+                        # If OMSLinuxAgent is installed, nudge the Azure Guest Agent to pickup the latest bits.
+                        $VMobj = Get-AzVM -ResourceGroupName $VM.ResourceGroupName -Name $VM.Name
+                        $omsExt = $VMobj.Extensions | where { ($_.Publisher -eq "Microsoft.EnterpriseCloud.Monitoring" -and $_.VirtualMachineExtensionType -eq "OmsAgentForLinux") }
+                        if ($omsExt -ne $null)
+                        {
+                            # Trigger a goal state change for GA.
+                            Set-AzVMExtension -VMName $VM.Name -ResourceGroupName $VM.ResourceGroupName -Location $VM.Location -Publisher $omsExt.Publisher -Name $omsExt.Name -ExtensionType $omsExt.VirtualMachineExtensionType
+                            Write-Host -ForegroundColor Green `t`t $VM.Name ": Set OMSLinuxAgent extension goal state for update"
```

Comment:
This will report success regardless of exit code of ^

Reply:
Yes, that's by design. Now, with -NoWait; we'd like to fire an forget..
The customer will run the script 2nd time (after 10 mins or so), to validate all are green.